scripts/install_cli.sh

```diff
@@ -1,63 +1,88 @@
 #!/bin/sh
-
 set -e
 
+# Install Kardinal CLI - supports bash, zsh, fish and assumes you have curl procps installed
+
+# Function to handle errors
+handle_error() {
+	echo "Ops! Failed to setup integration with your shell. Please add the following lines to
+your shell configuration manually (changes may not be persistent)
+export PATH=\$PATH:$BIN_FOLDER
+source <($BIN_FOLDER/$BINARY_NAME completion $PARENT_SHELL)"
+	exit 1
+}
+
+# Rest of your script goes here
 REPO="kurtosis-tech/kardinal"
 BINARY_NAME="kardinal"
-
 OS=$(uname -s | tr '[:upper:]' '[:lower:]')
 ARCH=$(uname -m)
-
-if [[ "$ARCH" == "x86_64" ]]; then
-	ARCH="amd64"
-elif [[ "$ARCH" == "arm64" ]]; then
-	ARCH="arm64"
-elif [[ "$ARCH" == "aarch64" ]]; then
-	ARCH="arm64"
-fi
+case "$ARCH" in
+x86_64) ARCH="amd64" ;;
+arm64 | aarch64) ARCH="arm64" ;;
+esac
 
 BIN_FOLDER="$HOME/.local/bin"
-mkdir -p $BIN_FOLDER
+mkdir -p "$BIN_FOLDER"
 
-LATEST_RELEASE=$(curl -s "https://api.github.com/repos/$REPO/releases/latest" | grep '"tag_name":' | sed -E 's/.*"([^"]+)".*/\1/')
+WAS_INTALLED_BEFORE=0
+if [ -f "$BIN_FOLDER/$BINARY_NAME" ]; then
+	WAS_INTALLED_BEFORE=1
+fi
 
+LATEST_RELEASE=$(curl -s "https://api.github.com/repos/$REPO/releases/latest" | grep '"tag_name":' | sed -E 's/.*"([^"]+)".*/\1/')
 echo "Downloading $BINARY_NAME $LATEST_RELEASE for $OS $ARCH..."
 DOWNLOAD_URL="https://github.com/$REPO/releases/download/$LATEST_RELEASE/${BINARY_NAME}-${OS}-${ARCH}"
-curl -L $DOWNLOAD_URL -o /$BIN_FOLDER/$BINARY_NAME
-chmod +x $BIN_FOLDER/$BINARY_NAME
-
-USER_SHELL=$(basename $SHELL)
-echo "Detected shell: $USER_SHELL"
-
-handle_error() {
-	local exit_code=$?
-	echo "Ops! Failed to setup integration with your $USER_SHELL shell. Please add the following lines to 
-  your shell configuration manually (changes may not be persistent)
-export PATH=\$PATH:$BIN_FOLDER
-source <($BIN_FOLDER/$BINARY_NAME completion $USER_SHELL)"
-	exit $exit_code
-}
+curl -L "$DOWNLOAD_URL" -o "$BIN_FOLDER/$BINARY_NAME"
+chmod +x "$BIN_FOLDER/$BINARY_NAME"
 
-trap 'handle_error' ERR
+PARENT_SHELL=$(ps -o comm= -p $PPID)
```

Comment:
i dont think this method of shell detection is reliable/universal, e.g. will fail on busybox systems (e.g. alpine) and will also error out in a docker environment i think since $PPID will be 0, but maybe thats an optimization for another time?

Reply:
this also detects my shell as `-zsh` instead of `zsh`, not sure what that is about?

Reply:
Do you have a more concrete example? I've tested on different docker containers (and the bash one is alpine):
`docker container run -it --rm dideler/fish-shell`
`docker container run -it --rm bash`
`docker container run -it --rm zshusers/zsh:5.9`

Reply:
Added the case for `-zsh`.
Merging as it is and we can improve it from this point.